HLTriggerOffline/Scouting/plugins/PatElectronTagProbeAnalyzer.cc

```diff
@@ -0,0 +1,843 @@
+    return;
+  }
+
+  edm::Handle<std::vector<Run3ScoutingElectron>> sctEls;
+  iEvent.getByToken(scoutingElectronCollection_, sctEls);
+  if (sctEls.failedToGet()) {
+    edm::LogWarning("ScoutingMonitoring") << "Run3ScoutingElectron collection not found.";
+    return;
+  }
+
+  edm::Handle<edm::ValueMap<bool>> tight_ele_id_decisions;
+  iEvent.getByToken(eleIdMapTightToken_, tight_ele_id_decisions);
+
+  edm::LogInfo("ScoutingMonitoring") << "Process pat::Electrons: " << patEls->size();
+  edm::LogInfo("ScoutingMonitoring") << "Process Run3ScoutingElectrons: " << sctEls->size();
+
+  // Trigger
+  edm::Handle<edm::TriggerResults> triggerResults;
+  iEvent.getByToken(triggerResultsToken_, triggerResults);
+  const edm::TriggerNames& triggerNames = iEvent.triggerNames(*triggerResults);
+  bool fire_singlePhoton_DST = hasPatternInHLTPath(triggerNames, "DST_PFScouting_SinglePhotonEB");
+  bool fire_doubleEG_DST = hasPatternInHLTPath(triggerNames, "DST_PFScouting_DoubleEG");
+
+  // Trigger Object
+  edm::Handle<pat::TriggerObjectStandAloneCollection> triggerObjects;
+  iEvent.getByToken(triggerObjects_, triggerObjects);
+
+  std::vector<std::string> filterToMatch = {"hltDoubleEG11CaloIdLHEFilter", "hltEG30EBTightIDTightIsoTrackIsoFilter"};
+  size_t numberOfFilters = filterToMatch.size();
+  trigger::TriggerObjectCollection* legObjects = new trigger::TriggerObjectCollection[numberOfFilters];
+  for (size_t iteFilter = 0; iteFilter < filterToMatch.size(); iteFilter++) {
+    std::string filterTag = filterToMatch.at(iteFilter);
+    for (pat::TriggerObjectStandAlone obj : *triggerObjects) {
+      obj.unpackNamesAndLabels(iEvent, *triggerResults);
+      if (obj.hasFilterLabel(filterTag)) {
+        legObjects[iteFilter].push_back(obj);
+      }
+    }
+  }
+
+  std::vector<int> sctElectron_gsfTrackIndex;
+  for (const auto& sct_el : *sctEls) {
+    size_t gsfTrkIdx = 9999;
+    bool foundGoodGsfTrkIdx = scoutingElectronGsfTrackIdx(sct_el, gsfTrkIdx);
+    if (foundGoodGsfTrkIdx)
+      sctElectron_gsfTrackIndex.push_back(gsfTrkIdx);
+    else
+      sctElectron_gsfTrackIndex.push_back(-1);
+  }
+
+  // for (const auto& pat_el : *patEls){
+  for (size_t i = 0; i < patEls->size(); ++i) {
+    const auto pat_el = patEls->ptrAt(i);
+    if (!((*tight_ele_id_decisions)[pat_el]))
+      continue;
+
+    ROOT::Math::PtEtaPhiMVector tag_pat_el(pat_el->pt(), pat_el->eta(), pat_el->phi(), pat_el->mass());
+    for (size_t j = 0; j < patEls->size(); ++j) {
+      //    for (const auto& pat_el_second : *patEls){
+      const auto pat_el_second = patEls->ptrAt(j);
+      if (i == j)
+        continue;
+      if (!((*tight_ele_id_decisions)[pat_el_second]))
+        continue;
+      ROOT::Math::PtEtaPhiMVector probe_pat_el(
+          pat_el_second->pt(), pat_el_second->eta(), pat_el_second->phi(), pat_el_second->mass());
+      float invMass = (tag_pat_el + probe_pat_el).mass();
+      if ((80 < invMass) && (invMass < 100)) {
```

Comment:
hardcoding the boundaries is bad practice, either make them configurable or make static constants.